src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -113,7 +113,7 @@ export class VSCodeKernelPickerProvider implements INotebookKernelProvider {
         }
         const [preferredKernel, kernels] = await Promise.all([
             this.getPreferredKernel(document, token, sessionManager),
-            this.getKernelSelections(document, token, sessionManager)
+            this.getKernelSelections(document, token)
         ]).finally(() => (sessionManager ? sessionManager.dispose() : undefined));
```

Comment:
The problem was, the list of kernels was still being fetched & sessionManager was disposed.
This is possible as we were lazy loading the kernels.
We return cached items immediately  & fetch in the background using the session manager.
But this code sees that the result has returned hence we dispose the session manager, & the background code falls over.